### PR TITLE
chore: update eth-account and hexbytes

### PR DESCRIPTION
**Pull request by DanielSchiavini**
This PR updates eth-account. Note that this version has a few [breaking changes](https://eth-account.readthedocs.io/en/latest/release_notes.html) that are not used in this project.

### What I did
- Updated eth-account
- Updated hexbytes (dependency of eth-account)

### How I did it
Updating the requirements in setup.py

### How to verify it
Tests should still pass

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] ~~New test cases have been added and are passing~~
- [x] Documentation has been updated
- [x] PR title follows

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]
Have to verify this update upstream to Ape to see if it should be a patch or minor breaking change

**Review by fubuloubu — DISMISSED** [review]

File: tests/test_common.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import pytest
-from eth_account._utils.structured_data.hashing import hash_message
+from eth_account.messages import encode_typed_data, _hash_eip191_message
```
Comment (DanielSchiavini) [inline tests/test_common.py:2]:
```suggestion
```

**Comment by antazoey** [comment]
> Have to verify this update upstream to Ape to see if it should be a patch or minor breaking change

I dont think Ape would be able to download this yet, until web3.py 7.0 is released, but I could be wrong.

**Comment by antazoey** [comment]
some parallels https://github.com/ApeWorX/eip712/pull/45

**Comment by fubuloubu** [comment]
@DanielSchiavini we are merging and releasing #45, please rebase your work off of that (if some changes are still available)

We can then release a minor breaking change

**Comment by DanielSchiavini** [comment]
@fubuloubu I don't think there's anything left after #45. Thank you!

**Pull request closed without merging**